tests/integration/learning/test_learning_versions.py

```diff
@@ -0,0 +1,99 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import pytest
+
+from constant_sorrow.constants import UNKNOWN_VERSION
+
+from nucypher.characters.lawful import Ursula
+from nucypher.config.constants import TEMPORARY_DOMAIN
+from nucypher.network.nodes import Teacher
+
+ursulas_v1 = (
```

Comment:
Would be useful to have a comment describing how these values were generated.

Reply:
What if I told you I just ran a debugger in some test that uses the `blockchain_ursulas` fixture, use an interactive console at some breakpoint, and just copy-pasted the output of `print(bytes(ursula).hex())`? yeah I know 🙈 🙈 🙈

Reply:
Well then I'd probably tell you can you don't need to involve a debugger for that, and it can just be made into a script and put in a comment :)

Reply:
✅ 
I introduced a "script" that simply uses the `blockchain_ursulas` fixture and print the proper hex-strings. I say "script" because it's actually a test, since I want to reuse some pytest fixtures and I don't want to duplicate any logic. Still a bit sketchy, but sufficient IMO, given how uncommon will be to bump the learning loop version.